### PR TITLE
Restructured code to reduce the number of allocations during dispatch.

### PR DESCRIPTION
**Pull request by nirvdrum**
I use Tree Style Tab with a fairly large number of tabs. I love the extension, but it causes my battery to drain too quickly. I'm sure at a lower number of tabs, this isn't much of a problem. But, I'm profiling it now and trying to fix some stuff.

This set of changes is fairly small, but cuts the retained memory for 18% -> 3% and the allocated memory from 8% -> 2% when I profiled with 25 tabs. I tried to profile under similar conditions and I only focused on "Web Extensions" in the profiler, but there are system variances and workload variances that likely impact it. I ran the profiles multiple times and got similar results each time. But, I'd still take the absolute numbers with a grain of salt and focus more on the relative difference between them. While I didn't focus explicitly on it, it appears from the profiles that GC frequency dropped as well.

The summary of changes are:
  * Eagerly allocate the `results` array with the correct size to avoid any array resizes
  * Instead of cloning the set of listeners just to `map` over the array, iterate over the set directly
    * I'm assuming either the iteration order doesn't matter or that the system will consistently iterate over a set the same way it would construct an array from a set. If iteration order is important, we should rethink storing the listeners as a set in the first place
    * This eliminates the array allocation and the data copy
    * This eliminates the allocation of the anonymous function passed to the `map` call

While I realize this changes from a functional style to a more procedural style, the performance impact is fairly large. I'm open to other approaches as well. I profiled keeping the `map` but hoisting the function passed to it so we can just pass a function reference rather an allocate a function object. That helps as well and maybe makes the code a bit cleaner, but the benefit was only about half of what this PR proposes.

### CODE REVIEW

File: EventListenerManager.js [hunk 1]
```diff
@@ -45,22 +45,25 @@ export default class EventListenerManager {
       return results.then(this.normalizeResults);
     else
       return this.normalizeResults(results);
-  }
+  };
```
Comment (piroor) [inline EventListenerManager.js]: This change looks needless. Could you revert only this part?

**Comment by piroor** [comment]
Thank you for investigation! I agree that my codes are often not optimized well, so optimizations are always welcome for me.

I'll merge this after the first needless part is reverted.

File: EventListenerManager.js (hunk 1 above)
Comment (nirvdrum) [reply EventListenerManager.js]: Sorry about that. I try not to change code I don't need to. I think I ran a code formatter to tidy up and forgot to drop this change. It's fixed now.

Comment (piroor) [reply EventListenerManager.js]: Thanks!

**Comment by piroor** [comment]
Thanks a lot! I'll update depender addons ASAP.

**Pull request merged**